istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,510 @@
+		Example: `
+# Disable the webhook configuration of Galley
+istioctl experimental post-install webhook disable --validation --config-name istio-galley
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !opts.disableValidationWebhook {
+				fmt.Println("not disabling validation webhook")
+				return nil
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				fmt.Printf("err when creating k8s client interface: %v\n", err)
+				return err
+			}
+			err = deleteValidatingWebhookConfig(client, opts.validatingWebhookConfigName)
+			if err != nil {
+				fmt.Printf("error when deleting validatingwebhookconfiguration: %v\n", err)
+				return err
+			}
+			fmt.Printf("validatingwebhookconfiguration %v has been deleted\n", opts.validatingWebhookConfigName)
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.disableValidationWebhook, "validation", true, "Specifies whether disabling"+
+		"the validating webhook.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "config-name", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to disable.")
+
+	return cmd
+}
+
+func newStatusCmd(opts *statusCliOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "status",
+		Short: "Get webhook configurations",
+		Example: `
+# Display the webhook configuration of Galley
+istioctl experimental post-install webhook status --validation --config-name istio-galley
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !opts.validationWebhook {
+				fmt.Println("not displaying validation webhook")
+				return nil
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				fmt.Printf("err when creating k8s client interface: %v\n", err)
+				return err
+			}
+			config, err := getValidatingWebhookConfig(client, opts.validatingWebhookConfigName)
+			if err != nil {
+				fmt.Printf("error getting validatingwebhookconfiguration: %v\n", err)
+				return err
+			}
+			b, err := yaml.Marshal(config)
+			if err != nil {
+				fmt.Printf("error when marshaling validatingwebhookconfiguration to yaml: %v\n", err)
+				return err
+			}
+			fmt.Printf("validatingwebhookconfiguration %v is:\n%v\n", opts.validatingWebhookConfigName, string(b))
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.validationWebhook, "validation", true, "Specifies whether displaying"+
+		"the validating webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "config-name", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to display.")
+
+	return cmd
+}
+
+// Create the validatingwebhookconfiguration
+func createValidatingWebhookConfig(k8sClient kubernetes.Interface,
+	config *v1beta1.ValidatingWebhookConfiguration) (*v1beta1.ValidatingWebhookConfiguration, error) {
+	var whConfig *v1beta1.ValidatingWebhookConfiguration
+	var err error
+	if config == nil {
+		return nil, fmt.Errorf("validatingwebhookconfiguration is nil")
+	}
```

Comment:
You control the calling code. You can probably skip the `config == nil` check.

Reply:
Done.